client.go

```diff
@@ -69,151 +63,50 @@ func WithPlatform(platform string) Option {
 	}
 }
 
+// GetImage parses the user provided image string and provides an image object;
+// note: the source where the image should be referenced from is automatically inferred.
+func GetImage(ctx context.Context, imgStr string, options ...Option) (*image.Image, error) {
+	return GetImageFromSource(ctx, imgStr, "", options...)
```

Comment:
I see it's convenient to allow for an empty string to be passed and it stands in for 'any', but given the function is called `GetImageFromSource` and should be distinctly different from `GetImage` in this particular functionality, I think a third unexported function should really be created that does the common functionality and the empty source checks end up being an explicit guard in the `GetImageFromSource` function.